scripts/doc_checker.py

```diff
@@ -0,0 +1,538 @@
+
+
+def monkeypatching():
+    """Monkeypatch decorators which change __doc__ attribute."""
+    import ray
+    import modin.utils
+
+    ray.remote = lambda *args, **kwargs: args[0]
+
+    modin.utils._inherit_docstrings = lambda *args, **kwargs: lambda cls: cls
+    modin.utils._inherit_func_docstring = lambda *args, **kwargs: lambda func: func
+
+
+def validate(
+    paths: List[pathlib.Path], add_ignore: List[str], use_numpydoc: bool
+) -> bool:
+    """
+    Perform pydocstyle and numpydoc checks.
+
+    Parameters
+    ----------
+    paths : List[pathlib.Path]
+        Filenames of directories for check.
+    add_ignore : List[str]
+        `pydocstyle` error codes which are not verified.
+    use_numpydoc : bool
+        Determine if numpydoc checks are needed.
+
+    Returns
+    -------
+    is_successfull : bool
+        Return True if all checks are successful.
+    """
+    is_successfull = True
+    for path in paths:
+        if not pydocstyle_validate(path, add_ignore):
+            is_successfull = False
+        if use_numpydoc:
+            if not numpydoc_validate(path):
+                is_successfull = False
+    return is_successfull
+
+
+def check_args(args: argparse.Namespace):
+    """
+    Check the obtained values for correctness.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Parser arguments.
+
+    Raises
+    ------
+    ValueError
+        Occurs in case of non-existent files or directories.
+    """
+    for path in args.paths:
+        if not path.exists():
+            raise ValueError(f"{path} does not exist")
+        abs_path = os.path.abspath(path)
+        if not abs_path.startswith(MODIN_PATH):
+            raise ValueError(
+                f"it is unsupported to use this script on files from another "
+                f"repository; script' repo '{MODIN_PATH}', "
+                f"input path '{abs_path}'"
+            )
+
+
+def get_args() -> argparse.Namespace:
+    """
+    Get args from cli with validation.
+
+    Returns
+    -------
+    argparse.Namespace
+    """
+    parser = argparse.ArgumentParser(
+        description="Check docstrings by using pydocstyle and numpydoc"
+    )
+    parser.add_argument(
+        "paths",
+        nargs="+",
+        type=pathlib.Path,
+        help="Filenames or directories; in case of direstories perform recursive check",
+    )
+    parser.add_argument(
+        "--add-ignore",
+        nargs="*",
+        default=[],
+        help="Pydocstyle error codes; for example: D100,D100,D102",
+    )
+    parser.add_argument(
+        "--disable-numpydoc",
+        default=False,
+        action="store_true",
+        help="Determine if numpydoc checks are not needed",
+    )
+    args = parser.parse_args()
+    check_args(args)
+    return args
+
+
+if __name__ == "__main__":
+    args = get_args()
+    monkeypatching()
+    if not validate(args.paths, args.add_ignore, not args.disable_numpydoc):
+        print("NOT SUCCESSFUL CHECK")
```

Comment:
A couple of questions here:

* Should this go to stderr instead of stdout?
* Should we show which files are incorrect? I think that would help a lot.
* A minor rewording suggestion:

```suggestion
        print("Invalid Documentation Found")
```

Reply:
> A couple of questions here:
> 
> * Should this go to stderr instead of stdout?

I think that errors found in the documentation should go to `stdout`, and errors related to the operation of the script itself should go to `stderr`. That is, in the case of this print, it will be output to the `stdout` with "error" mark.


> * Should we show which files are incorrect? I think that would help a lot.

The output of this script already displays the filepaths. For example:

```python
..\projects\modin>python scripts\doc_checker.py asv_bench\benchmarks\utils.py

INFO:PYDOCSTYLE OUTPUT FOR asv_bench\benchmarks\utils.py
ERROR:asv_bench\benchmarks\utils.py:125 in public class `weakdict`:
        D101: Missing docstring in public class

INFO:NUMPYDOC OUTPUT FOR asv_bench\benchmarks\utils.py - CAN BE EMPTY
ERROR:asv_bench.benchmarks.utils.translator_groupby_ngroups:102:PR01:Parameters {'groupby_ngroups'} not documented    
ERROR:asv_bench.benchmarks.utils.translator_groupby_ngroups:102:PR02:Unknown parameters {'groupby_ngroups: str or int'}
ERROR:asv_bench.benchmarks.utils.translator_groupby_ngroups:102:PR10:Parameter "groupby_ngroups" requires a space before the colon separating the parameter name and type
ERROR:asv_bench.benchmarks.utils.weakdict:125:GL08:The object does not have a docstring
ERROR:INVALID DOCUMENTATION FOUND
```
Are you ok with the output, or we need to add something else?